coderd/workspacebuilds.go

```diff
@@ -160,6 +161,70 @@ func (api *API) workspaceBuilds(rw http.ResponseWriter, r *http.Request) {
 	httpapi.Write(rw, http.StatusOK, apiBuilds)
 }
 
+func (api *API) workspaceBuildByBuildNumber(rw http.ResponseWriter, r *http.Request) {
```

Comment:
(blocking): Please add some corresponding unit tests for this new API route!

Reply:
Interestingly, our existing tests simulate the CLI, and don't directly test the method. Even with this PR, we are still using the old API (`builds/<build_id>`) for CLI. Do you think we should update the CLI to use the new API and then add tests? Meanwhile, I'll look for some references in the code to add tests for this usecase too.

Reply:
>  Do you think we should update the CLI to use the new API 

Is there any significant benefit to be had from doing this? If so, it would be better to open a new issue for that so it can be addressed separately.

> our existing tests simulate the CLI, and don't directly test the method

Not exactly - see https://github.com/coder/coder/blob/main/coderd/workspacebuilds.go#L22 : we test with the `codersdk.Client` method, which hits the API. It's _some_ of what the CLI tests do.

Reply:
>Is there any significant benefit to be had from doing this? If so, it would be better to open a new issue for that so it can be addressed separately.

Personally, I don't think so.

>Not exactly - see https://github.com/coder/coder/blob/main/coderd/workspacebuilds.go#L22 : we test with the `codersdk.Client` method, which hits the API. It's _some_ of what the CLI tests do.

Thanks, I was not sure if we'd write a client method only for tests. I can definitely try and give that a shot.

Reply:
Added tests!